lib/chef/run_context.rb

```diff
@@ -198,76 +337,152 @@ def resolve_attribute(cookbook_name, attr_file_name)
       attribute_filename
     end
 
-    # An Array of all recipes that have been loaded. This is stored internally
-    # as a Hash, so ordering is predictable.
     #
-    # Recipe names are given in fully qualified form, e.g., the recipe "nginx"
-    # will be given as "nginx::default"
+    # A list of all recipes that have been loaded.
+    #
+    # This is stored internally as a Hash, so ordering is predictable.
+    #
+    # TODO is the above statement true in a 1.9+ ruby world?  Is it relevant?
```

Comment:
In Ruby 2.0+ `Hashes enumerate their values in the order that the corresponding keys were inserted.` - http://ruby-doc.org/core-2.0.0/Hash.html

Your statement is true and this TODO can be removed.